lib/timers.js

```diff
@@ -301,6 +302,13 @@ const Timeout = function(after) {
   this._repeat = null;
 };
 
+
+function destroyOnTimeout() {
+  if (!this.owner._repeat)
+    this.owner.close();
+}
```

Comment:
Perhaps, but may be it should not be there at all. I'll figure out the proper way of doing it in a bit.